### ISSUE TITLE
Multi-parameter regression with custom loss function

### ISSUE THREAD

**Issue by musella** [dmlc/xgboost#2830]
We would like to use the XGB regressor with a loss function that depends on many parameters.

For example, we would like to simultaneously estimate mean and width of a gaussian pdf using the loss function:
loss = np.mean( (ytrue-ypred)/spred + 2.*log(spred) )

Here ypred and spred are the estimators of mean and width of the gaussian pdf. 

Is there a way to get the XGB regressor to fit for these parameters?

Thanks a lot.
Pasquale
